Lib/test/test_enum.py

```diff
@@ -203,59 +203,298 @@ class Holiday(date, Enum):
+
+    def test_mixin_dirs(self):
+        from datetime import date
+
+        enums_for_test = (
+            # generic mixins from enum.py
+            (IntEnum, int),
+            (StrEnum, str),
+            # generic mixins from outside enum.py
+            (self.FloatEnum, float),
+            (self.DateEnum, date),
+            # concrete mixin from enum.py
+            (self.Grades, int),
+            # concrete mixin from outside enum.py
+            (self.Holiday, date),
+            # concrete mixin from enum.py with added behaviour
+            (self.IntWowser, int),
+            # concrete mixin from outside enum.py with added behaviour
+            (self.FloatWowser, float)
+        )
+
+        enum_dict = Enum.__dict__
+        enum_dir = dir(Enum)
+        enum_module_names = enum.__all__
+        is_from_enum_module = lambda cls: cls.__name__ in enum_module_names
+        is_enum_dunder = lambda attr: enum._is_dunder(attr) and attr in enum_dict
+
+        # General tests
+        for enum_cls, mixin_cls in enums_for_test:
+            with self.subTest(enum_cls=enum_cls):
+                cls_dir = dir(enum_cls)
+                cls_dict = enum_cls.__dict__
+
+                mixin_attrs = [
+                    x for x in dir(mixin_cls)
+                    if (
+                        getattr(mixin_cls, x) is not getattr(object, x, object())
+                        and x not in {'__init_subclass__', '__subclasshook__'}
+                    )
+                ]
+
+                first_enum_base = next(
+                    base for base in enum_cls.__mro__
+                    if is_from_enum_module(base)
+                )
+
+                for attr in mixin_attrs:
+                    with self.subTest(attr=attr):
+                        if enum._is_sunder(attr):
+                            # Unlikely, but no harm in testing
+                            self.assertNotIn(attr, cls_dir)
+                        elif attr in {'__class__', '__doc__', '__members__', '__module__'}:
+                            self.assertIn(attr, cls_dir)
+                        elif is_enum_dunder(attr):
+                            if is_from_enum_module(enum_cls):
+                                self.assertNotIn(attr, cls_dir)
+                            elif getattr(enum_cls, attr) is getattr(first_enum_base, attr):
+                                self.assertNotIn(attr, cls_dir)
+                            else:
+                                self.assertIn(attr, cls_dir)
+                        else:
+                            self.assertIn(attr, cls_dir)
+
+        # Some specific examples
+        int_enum_dir = dir(IntEnum)
+        self.assertIn('imag', int_enum_dir)
+        self.assertIn('__rfloordiv__', int_enum_dir)
+        self.assertNotIn('__format__', int_enum_dir)
+        self.assertNotIn('__hash__', int_enum_dir)
+
+        class OverridesFormatOutsideEnumModule(Enum):
```

Comment:
Do we lose IDE or completion help by not listing `__format__` in `dir()`?

Reply:
No, we don't, and I agree that `__format__` should be excluded from the `dir()` of `Enum`, `IntEnum` and `StrEnum`, as well as any classes that inherit from these enum classes without overriding `__format__`. This is still the case in my PR.

But if a user is defining a custom `__format__` method in their own `Enum` class, I think it would be surprising for the user if `__format__` did not show up in `dir()` or `help()`. Their implementation of `__format__` could have some unusual or surprising behaviour, which they might explain in detail in the docstring; but the docstring is of limited use if it doesn't show up in the output of `help()`.